pkg/daemon/action.go

```diff
@@ -37,6 +39,20 @@ func (pi *policyInstall) do(modulePath string, sh semodule.Handler, ds datastore
 		return "", fmt.Errorf("failed getting a safe path for policy: %w", err)
 	}
 
+	cs, csErr := utils.Checksum(policyPath)
+	if csErr != nil {
+		return "", fmt.Errorf("installing policy: %w", err)
+	}
+
+	p, getErr := ds.Get(policyName)
```

Comment:
Have you considered ALSO always checksumming the contents on the disk? This could provide the integrity protection we talked about some time ago and ensure that wrong contents of the file are always reverted. I think the performance hit is really negligible, the bigger question is if selinuxd is the right place to do this.

Reply:
Let's discuss this and do this in a subsequent PR.

Reply:
But is the policy taken into account at any other time than when it's written into and compiled? I think the separate goroutine would have made more sense for seccomp policies that are not (AFAIK?) complied in, but for the selinux policies, the only entrypoint into taking them to use is the compile time.

I don't know how does the result of the policies look like, if there is a one binary blob per policy, then perhaps it would make sense to hash the binary policy and periodically check that. But for checking the source which lives anyway only in the selinuxd pod at the moment, I don't see much difference between checking the source at install time and periodically.

Reply:
Hmm, sorry, I was in the selinuxd-in-a-pod mindset. If selinuxd is running directly on a host, then it would be nice to also check the source.. Still, wouldn't it be nice that the result matches?  OTOH I don't know if different base policies or different libselinux would not result in different binaries...
So if we want to take into account the possibility of selinuxd running on the host then yes, the separate goroutine might be a good idea..

Reply:
Let's do this, let me open a GH issue so we keep that work in mind, and we can address that in a subsequent PR. what do you think?

Reply:
@jhrozek what about https://github.com/JAORMX/selinuxd/issues/33 ?